pkgs/applications/audio/bespokesynth/default.nix

```
@@ -58,28 +59,24 @@ let
 in
 stdenv.mkDerivation rec {
   pname = "bespokesynth";
-  version = "1.1.0";
+  version = "1.2.0";
 
   src = fetchFromGitHub {
     owner = "BespokeSynth";
     repo = pname;
-    rev = "v${version}";
-    sha256 = "sha256-PN0Q6/gI1PeMaF/8EZFGJdLR8JVHQZfWunAhOIQxkHw=";
+    rev = "71134ce432dd15ae773cd49ac7c6e436d37e8a46";
+    hash = "sha256-pEifqbzJYI0EnGRQL0a0hzNkUw4+ZE6mrMGUVngzTZY=";
```

I didn't pick up on the fact that when you force-pushed from bda87f7beedf33ce00bad49ad34025ccea373c02 to 179c1e2b601448d2a14f72218b8b17a0046cc5b8 , you went back to using an unstable commit without marking the `version` as such. 

https://github.com/BespokeSynth/BespokeSynth/commit/71134ce432dd15ae773cd49ac7c6e436d37e8a46 != https://github.com/BespokeSynth/BespokeSynth/releases/tag/v1.2.0 . 

If this is still supposed to be `version = "1.2.0"` then please revert the `rev` line to the format that was used before and change `hash` to what the actual hash of that version is:

```suggestion
    rev = "v${version}";
    hash = "sha256-baBZBRK8xbtnj/MJaTA036exLjvvucfrsk/Ue5oiG7c=";
```

---

Sorry, I should have left a comment saying I did so. I updated to the latest commit because I learned the stable version had broken jedi functionality (python autocomplete) on MacOS. Since I don't know how long it was going to be until the next release (or opportunity to update this package), I went ahead and updated to the latest commit (which fixed this). I skipped updating the version number because I saw it as basically being a hotfix to v1.2.0, but I should probably mark it as unstable. Should I mark it as `v1.2.0-unstable` or would the date of the commit be better?

---

It seems like most of the packages opted for `v1.2.0+unstable=2023-07-20`:
```
pkgs/tools/misc/tty-clock/default.nix
5:  version = "2.3+unstable=2021-04-07";

pkgs/tools/misc/dialogbox/default.nix
11:  version = "1.0+unstable=2020-11-16";

pkgs/applications/emulators/bsnes/higan/default.nix
25:  version = "115+unstable=2021-08-18";

pkgs/applications/emulators/emu2/default.nix
8:  version = "0.pre+unstable=2021-09-22";

pkgs/applications/graphics/gimp/plugins/default.nix
225:    version = "2.2+unstable=2021-12-03";

pkgs/applications/graphics/inkscape/extensions/applytransforms/default.nix
9:  version = "0.pre+unstable=2021-05-11";

pkgs/applications/terminal-emulators/st/siduck76-st.nix
15:  version = "0.pre+unstable=2021-08-20";

pkgs/applications/terminal-emulators/st/lukesmithxyz-st/default.nix
15:  version = "0.pre+unstable=2021-08-10";

…
```

---

> I learned the stable version had broken jedi functionality (python autocomplete) on MacOS

If it's easy to figure out which commits fixed this, then you could pull them in and apply them onto 1.2.0:

```nix
patches = [
  # Fixes jedi functionality on Darwin
  # Remove when version > 1.2.0
  (fetchpatch {
    url = "https://github.com/BespokeSynth/bespokesynth/commit/whatevercommitfixesthis.patch";
    hash = "sha256-whateverhashisneededhere=";
  })
];
```

> I skipped updating the version number because I saw it as basically being a hotfix to v1.2.0, but I should probably mark it as unstable

Unless it's an actual release with a tag (or in some other form actually treated by upstream as a release, with or without tag), it should be treated as an unstable commit instead of a release. For pulling in a fix into a release, see above.

> Should I mark it as `v1.2.0-unstable` or would the date of the commit be better?

The above would be preferable IMO. If you want to ditch the applying-patches way, then back to what [the manual](https://nixos.org/manual/nixpkgs/stable/#sec-package-naming) says unstable versions should use: `unstable-<YYYY-MM-DD of commit>`

---

> It seems like most of the packages opted for `v1.2.0+unstable=2023-07-20`

This versioning pattern is from [a not-yet-accepted RFC](https://github.com/NixOS/rfcs/pull/107), dates used in them is ~ when the RFC was first opened. And they don't align with what the RFC currently suggests to use: `3.0:date=2008-12-04`.

I'd prefer to wait until *any* consensus is reached on that RFC & it gets accepted before using its pattern, as right now it conflicts with what the manual has to say on this.

---

Okay! When I get the chance, I'll update the version to `unstable-2023-08-17` and the commit hash to [`c6b141`](https://github.com/BespokeSynth/BespokeSynth/commit/c6b1410afefc8b0b9aeb4aa11ad5c32651879c9f) (slightly behind `main`). I think that's a good stable stopping place because most things added after it are features.
Does this sound good?

---

I'll also ask awwbees about making that into `v1.2.1`, as I think that would be a good place for a new version.